src/structures/junction.cpp

```diff
@@ -27,11 +32,15 @@ bool operator<(Junction const & lhs, Junction const & rhs)
            : lhs.get_mate2() != rhs.get_mate2()
              ? lhs.get_mate2() < rhs.get_mate2()
              : lhs.get_inserted_sequence() < rhs.get_inserted_sequence();
+             // TODO (23.7.21, irallia): tandem_dup_count doesn't play a role here right?
```

Comment:
@joshuak94 @joergi-w If the amount of duplications is bigger or smaller between junctions, does not play a role for size comparsion right? Just for equality.

Reply:
That would be weird... 
In general, the following must hold: If a < b and b < a are both false, then a = b. 
This would be violated if you omit the duplication count.

Reply:
Aaah, you mean a duplication is larger if it contains more duplications?
And thus, one junction is larger than another if this is the case.
Right, but we look before whether a junction is smaller from the position perspective (mate < mate). The question is, does it matter how high the tandem_dup_count is and if so where would you put that? Is that more relevant than the length of the inserted sequence?

Reply:
I guess, I haven't yet fully understood what a junction is... 
Can two junctions with equal positions and sequence lengths possibly have a different duplication count? If yes, I would include it with the lowest priority.
My point is, if **any** of the fields in class `junction` differ, there must be a clear order (so you can only omit field comparisons that are indirectly determined, or that are omitted in the equality comparison as well).

Reply:
I have now thought about it again.
A junction describes a change in the read in comparison to the ref (e.g. insertion, duplication, deletion). It consists of two endpoints (breakends), a possibly inserted sequence and the name of the read and now also the number of duplications.
If we now compare two junctions, we want to know if they
1. share approximately the same positions (breakend 1 and 2), then
2. if they contain a similar number of duplications and
3. the inserted sequence is of similar length.

Important to note, for a novel element insertion the inserted sequence is the whole new sequence, for a duplication it is the duplicated sequence, this can be shorter if there are multiple duplications. Therefore, I now compare first on duplication and then on the sequence.
The content of the sequence is not compared yet, but that is still planned.